rskj-core/src/main/java/co/rsk/jsonrpc/JsonRpcIdentifiableMessage.java

```diff
@@ -22,28 +22,30 @@
 /**
  * The basic JSON-RPC request or response. It is required to have an ID.
  *
- * Note that the JSON-RPC 2.0 spec allows using strings as IDs, but our implementation doesn't.
+ * Note that the JSON-RPC 2.0 spec allows using strings as IDs.
  */
 public abstract class JsonRpcIdentifiableMessage extends JsonRpcMessage {
-    private final int id;
+	
+	private static final String ID_BAD_PARAMETER_MSG = "JSON-RPC message id should be a String or a positive number, but was %s.";
```

Comment:
There's nothing preventing Jackson from interpreting the string as a negative number. That should probably still be checked.